sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -55,6 +55,36 @@ def test_mean_shift():
     assert_equal(n_clusters_, n_clusters)
 
 
+def test_mean_shift_negative_bandwidth():
+    bandwidth = -1
+    ms = MeanShift(bandwidth=bandwidth)
+    msg = \
+        "bandwidth needs to be greater than zero or None," \
+        "            got -1.000000"
+    assert_raise_message(ValueError, msg, ms.fit, X)
+
+
+def test_seeds():
+    ms = MeanShift(seeds=None)
+    _ = ms.fit(X).labels_
+    assert_equal(ms.seeds is None, True)
+
+
+def test_mean_shift_cluster_all_false():
+    bandwidth = 1.2
+    ms = MeanShift(cluster_all=False)
+    labels = ms.fit(X).labels_
+    labels_unique = np.unique(labels)
+    n_clusters_ = len(labels_unique)
+    assert_equal(n_clusters_ > n_clusters, True)
+
+    cluster_centers, labels = mean_shift(X, bandwidth=bandwidth,
```

Comment:
Rather than repeat the code, please use pytest.mark.parameterize to test multiple settings of bandwidth

Reply:
changed to use 
pytest.mark.parameterize

Reply:
@jnothman please review

Reply:
@jnothman @ogrisel please review